src/ui/ui_settings/ui_settings_service.js

```diff
@@ -0,0 +1,130 @@
+import { defaultsDeep, noop } from 'lodash';
+import { errors as esErrors } from 'elasticsearch';
+
+import { getDefaultSettings } from './defaults';
+
+function hydrateUserSettings(userSettings) {
+  return Object.keys(userSettings)
+    .map(key => ({ key, userValue: userSettings[key] }))
+    .filter(({ userValue }) => userValue !== null)
+    .reduce((acc, { key, userValue }) => ({ ...acc, [key]: { userValue } }), {});
+}
+
+export class UiSettingsService {
+  constructor(options = {}) {
+    const {
+      index,
+      type,
+      id,
+      callCluster,
+      readInterceptor = noop,
+    } = options;
+
+    this._callCluster = callCluster;
+    this._readInterceptor = readInterceptor;
+    this._index = index;
+    this._type = type;
+    this._id = id;
+  }
+
+  getDefaults() {
+    return getDefaultSettings();
+  }
+
+  // returns a Promise for the value of the requested setting
+  async get(key) {
+    const all = await this.getAll();
+    return all[key];
+  }
+
+  async getAll() {
+    const raw = await this.getRaw();
+
+    return Object.keys(raw)
+      .reduce((all, key) => {
+        const item = raw[key];
+        const hasUserValue = 'userValue' in item;
+        all[key] = hasUserValue ? item.userValue : item.value;
+        return all;
+      }, {});
+  }
+
+  async getRaw() {
+    const userProvided = await this.getUserProvided();
+    return defaultsDeep(userProvided, this.getDefaults());
+  }
+
+  async getUserProvided(options) {
+    return hydrateUserSettings(await this._read(options));
+  }
+
+  async setMany(changes) {
+    await this._write(changes);
```

Comment:
Correct, I verified any nothing was consuming the resolved value of the "write" methods on this class so I just removed them, especially since they were useless.